frontend/jsdom-polyfill-env.js

```diff
@@ -0,0 +1,19 @@
+const Environment = require("jest-environment-jsdom")
+
+/**
+ * A custom environment to set the TextEncoder and TextDecoder that are required
+ * by arrow-js
+ */
+module.exports = class CustomTestEnvironment extends Environment {
```

Comment:
This file's purpose is non-obvious to me (but it may just be my lack of frontend experience). Is the polyfill used in tests only, or is it also used in the runtime? Can you add a few more sentences explaining the issue it's fixing?